swap/swap.go

```diff
@@ -309,19 +323,17 @@ func (s *Swap) loadCheque(peer enode.ID) (err error) {
 	return
 }
 
-//Clean up Swap
-func (swap *Swap) Close() {
-	swap.stateStore.Close()
+// Close cleans up swap
+func (s *Swap) Close() {
+	s.stateStore.Close()
 }
 
 // resetBalance is called:
 // * for the creditor: on cheque receival
 // * for the debitor: on confirmation receival
-func (s *Swap) resetBalance(peerID enode.ID) {
-	//TODO: reset balance based on actual amount
-	//TODO: review the locks
-	log.Info(fmt.Sprintf("resetting balance for peer %s", peerID.String()))
-	s.balances[peerID] = 0
+func (s *Swap) resetBalance(peerID enode.ID, amount int64) {
```

Comment:
Just a thought: why is this function called `resetBalance` rather than `updateBalance`?

Reply:
I thought it would be nice to have it explicit - `updateBalance` could be an update from anywhere in the code with any amount, while `resetBalance` would semantically express that we are resetting it after creating a cheque. A reset IMHO does not necessarily mean to zero, it can just mean "cancel out the difference" or something like that.

Having said that, `resetBalance` is an inmediate indirection to `updateBalance`, so it is of course not really needed.

Reply:
Can't argue with that logic.